src/groupByKey.ts

```diff
@@ -1,7 +1,18 @@
-type GroupsMap<T> = {
-  [key: string]: T[];
-};
-
-export function groupByKey(items, key) {
+export function groupByKey<T>(items: T[], key: keyof T): object {
   // write code here;
+  const result: any = {};
+
+  items.forEach((a: T) => {
+    const variable: any = a[key];
+
+    if (Object.keys(result).includes(String(variable))) {
```

Comment:
You can check it using `in` or `hasOwnProperty`

Reply:
Thanks)